src/input.cpp

```diff
@@ -225,3 +245,109 @@ void ReplayInput::late_update() {
 [[nodiscard]] bool ReplayInput::is_end_of_recording() const {
     return m_next_record_index >= m_recording_reader->num_records();
 }
+
+
+#if defined(__ANDROID__)
+void TouchInput::handle_event(const SDL_Event& event) {
+    const auto input_event = sdl_event_to_input_event(event);
+    if (input_event.has_value()) {
+        Input::handle_event(*input_event);
+    }
+}
+
+tl::optional<InputEvent> TouchInput::sdl_event_to_input_event(const SDL_Event& event) {
+    //TODO to handle those things better, holding has to be supported
+
+
+    // also take into accounts fingerId, since there may be multipel fingers, each finger has it's own saved state
+    const SDL_FingerID finger_id = event.tfinger.fingerId;
+
+    // this is used, to get the percentage, since it' all constexpr it's mainly for the developer to not think about percentages but about a pixel range on a certain device, but then it works as expected everywhere
+    constexpr auto screen_h_reference = 2160.0;
+    constexpr auto screen_w_reference = 1080.0;
```

Comment:
Should this be dependend on the device type or is this the "logical render resolution" of SDL?